google/cloud/internal/opentelemetry.h

```diff
@@ -108,12 +109,28 @@ Status EndSpan(opentelemetry::trace::Span& span, Status const& status);
  * composition.
  */
 template <typename T>
```

Comment:
Why declare the template immediately before defining it?

Reply:
Because I don't know how to C++. Removed.